### DISCUSSION TITLE
how to avoid Reached gathering budget of 5s warning in lighthouse

### DISCUSSION THREAD

**Discussion by Haris-Asghar** [GoogleChrome/lighthouse#15528] [Q&A]
how to avoid this warning in lighthouse:

 LH:artifacts:getArtifact ImageElements +35ms LH:ImageElements:warn Reached gathering budget of 5s. Skipped extra details for 281/314 +6s

and I have following code for generating lighthouse audit report:

async function runLighthouse(url, outputPath, iterations) {
    const chrome = await chromeLauncher.launch({ chromeFlags: ["--headless"] });

    for (let i = 0; i < iterations; i++) {
        const options = {
            // disableStorageReset: true, // Disable clearing the browser storage between runs
            logLevel: "info",
            output: "json",
            port: chrome.port,
            throttling: {
                method: "devtools",
                devtools: {
                    networkThrottling: {
                        downloadThroughputKbps: 1000,
                    },
                    cpuThrottling: 4, // Simulate slower CPU for more consistent results
                },
            },
            emulatedFormFactor: 'desktop',
        };
        const runnerResult = await lighthouse(url, options);

        const filename = `${outputPath}/${i + 1}.json`;
        const reportJson = runnerResult.report;
        fs.writeFileSync(filename, reportJson);
        console.log(`Lighthouse report saved to ${filename}`);
    }

    await chrome.kill();
}

**Comment by adamraine** [comment]
Related issue: https://github.com/GoogleChrome/lighthouse/issues/11289

314 images is a lot of images for Lighthouse to collect extra data on so it makes sense that the budget cut it off at 5s. Extrapolating from the 281/384 figure given, it would take ~40-50s just to collect the extra image data on this page.

There isn't really a way to prevent this warning unfortunately.
